### DISCUSSION TITLE
what is the meaning of  jacobians argument in exp_map and log_map functions ?

### DISCUSSION THREAD

**Discussion by Christopher6488** [facebookresearch/theseus#289] [Q&A] [answered]
## ❓ Questions and Help
what is the meaning of  jacobians argument in exp_map and log_map functions ?

**Comment by luisenp** [comment]
Hi @Christopher6488. Are you referring to the meaning in terms of API or in terms of geometry? cc @fantaosha

**Comment by fantaosha** [comment, marked as answer]
Hi @Christopher6488 , thanks for your question. We use the right trivialized tangent map and its inverse as the jacobians for `exp_map` and `log_map`, respectively. The jacobian, i.e., right trivialized tangent map, of the `exp_map` is defined as `Adjoint(exp(xi))^-1 * Dexp(xi)` where `Dexp(xi)` computes the the derivative of `exp(xi)` on `xi`. The jacobian, i.e., the inverse of right trivialized tangent map, of the `log_map` is defined as `Dlog(g) * Adjoint(g)` where `Dlog(g)` is the Euclidean derivative of `log(g)` on `g` as a real matrix. A more detailed explanation can be found in Eqs. 5 to 7 in https://asco.lcsr.jhu.edu/papers/KoMa2010.pdf
